### PR TITLE
[android] Switch to new single-header modulemap for Bionic too

### PR DESCRIPTION
**Pull request by finagolfin**
This pull gets all the same tests to pass in the Swift compiler test and validation suite when natively run on Android in [the Termux app](https://termux.com/), but also gets 38 C++ interopability tests that were previously failing to pass, as was the intent of #32404. However, building `libFoundationNetworking.so` from the corelibs then fails with these errors:
```
<unknown>:0: error: fatal error encountered while reading from module 'Foundation'; please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project
<unknown>:0: note: module 'Foundation' full misc version is '5.4(5.4)/Swift version 5.4-dev (LLVM e2976fe639d1f50, Swift f9be289b9e04b3d)'

*** DESERIALIZATION FAILURE (please include this section in any bug report) ***
top-level value not found
Cross-reference to module 'CDispatch'
... memcmp
... with type (UnsafeRawPointer, UnsafeRawPointer, Int) -> Int32
--------
1.      Swift version 5.4-dev (LLVM e2976fe639d1f50, Swift f9be289b9e04b3d)
2.      While evaluating request ExecuteSILPipelineRequest(Run pipelines { PrepareOptimizationPasses, EarlyModulePasses, HighLevel,Function+EarlyLoopOpt, HighLevel,Module+StackPromote, Serialize, MidLevel,Function, ClosureSpecialize, LowLevel,Function, LateLoopOpt, SIL Debug Info Generator } on SIL for FoundationNetworking.FoundationNetworking)
3.      While running pass #1974 SILModuleTransform "PerformanceSILLinker".
4.      While deserializing SIL function "$s10Foundation4DataV2eeoiySbAC_ACtFZ"
5.      While deserializing SIL function "memcmp"
<unknown>:0: error: unable to execute command: Aborted
```
I am not familiar with this error, so maybe a Swift compiler dev can chime in. I can reproduce when cross-compiling the corelibs for Android with #33724 and a couple other related pulls.

@CodaFi, @drodriguez, @hlopko, any input would be appreciated.

### CODE REVIEW

File: stdlib/public/Platform/SwiftGlibc.h.gyb
Comment (buttaface) [inline stdlib/public/Platform/SwiftGlibc.h.gyb:31]: This header is included twice, so removed one.

**Comment by buttaface** [comment]
I just tried this pull again with the May 3rd trunk source snapshot and got the following error this time:
```
<unknown>:0: error: fatal error encountered while reading from module 'Foundation'; please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project
<unknown>:0: note: module 'Foundation' full misc version is '5.5(5.5)/Swift version 5.5-dev (LLVM 6337fa4dc365079, Swift a6915e7bfef8173)'

*** DESERIALIZATION FAILURE (please include this section in any bug report) ***
top-level value not found
Cross-reference to module 'CDispatch'
... memcmp
... with type (UnsafeRawPointer, UnsafeRawPointer, Int) -> Int32
Notes:
* 'memcmp' in module 'SwiftShims' was filtered out.
* 'memcmp' in module 'Foundation' was filtered out.
* 'memcmp' in module 'CDispatch' was filtered out.
* 'memcmp' in module 'Glibc' was filtered out.
* 'memcmp' in module 'SwiftGlibc' was filtered out.
* 'memcmp' in module 'SwiftOverlayShims' was filtered out.
* 'memcmp' in module 'CoreFoundation' was filtered out.
* 'memcmp' in module 'CFURLSessionInterface' was filtered out.
----------
1.	Swift version 5.5-dev (LLVM 6337fa4dc365079, Swift a6915e7bfef8173)
2.	While evaluating request ExecuteSILPipelineRequest(Run pipelines { PrepareOptimizationPasses, EarlyModulePasses, HighLevel,Function+EarlyLoopOpt, HighLevel,Module+StackPromote, Serialize, MidLevel,Function, ClosureSpecialize, LowLevel,Function, LateLoopOpt, SIL Debug Info Generator } on SIL for FoundationNetworking.FoundationNetworking)
3.	While running pass #1640 SILModuleTransform "PerformanceSILLinker".
4.	While deserializing SIL function "$s10Foundation4DataV2eeoiySbAC_ACtFZ"
5.	While deserializing SIL function "memcmp"
<unknown>:0: error: unable to execute command: Aborted
```
@xymus, you added these new messages about `Glibc` being wrongly filtered out in #36431, any clue what might be going on or how to fix it?

**Comment by buttaface** [comment]
Alright, I will try that.

**Comment by buttaface** [comment]
Rebased and that last `malloc` issue appears to have gone away. Applying this patch to the Nov. 19 trunk source snapshot got 50 C++ Interop tests that were failing to pass natively on Android, while natively compiling the rest of the toolchain now works. I then applied this patch to cross-compiling the last Dec. 6 trunk snapshot for Android, buttaface/swift-android-sdk@e174127, and was able to [cross-compile the SDK and 11 Swift packages that passed their tests on the Android x86_64 emulator](https://github.com/buttaface/swift-android-sdk/runs/4514323126?check_suite_focus=true).

I had to patch Swift NIO for some Bionic constants in that commit, but that's it. Hopefully, I won't need to [revert the C++ interoperability that @zoecarver added to libswift any longer](https://github.com/buttaface/termux-packages/blob/swift56/packages/swift/swift-libswift-revert-libcxx-bindings.patch) once this pull is in.

@MaxDesiatov, would you run the CI?

@xymus, would you review?

@drodriguez, you may want to try this on your local setup that reproduces the Android CI before this is merged, though I doubt it will break that CI.

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Review by vgorloff — APPROVED** [review]
PR changes working on my side, when applied on top of toolchain https://github.com/vgorloff/swift-everywhere-toolchain/releases/tag/1.0.77.

The only thing I changed – is maintained C/C++ header search paths (passed to `swift build` command).

```
# Before
-Xcc -I/usr/local/ndk/21.4.7075529/.../include/c++/v1

# After
-Xcxx -I/usr/local/ndk/21.4.7075529/.../include/c++/v1.
```

**Review by gottesmm — APPROVED** [review]
If it works for all of you, LGTM

**Comment by buttaface** [comment]
@drodriguez, I just ran the compiler validation suite on this pull and #40625 for both AArch64 and armv7 and the same tests passed as on the Android CI, ie only the 13 C++ Interop tests that are currently broken on the Android CI failed (I was hoping this patch would fix those 13 tests, but no such luck).

I think you can merge this now, shouldn't break anything.

**Comment by buttaface** [comment]
@artemcm, would you run the CI again? It hasn't been re-run since the holiday break.

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by buttaface** [comment]
Ping, @drodriguez, just waiting for you to merge this.

**Comment by buttaface** [comment]
@CodaFi, would you merge? This will unbreak the build on the Android CI, and if @drodriguez has any changes he wants made, we can always add those whenever he gets back.

**Comment by buttaface** [comment]
@drodriguez, can we get this in?

**Comment by buttaface** [comment]
Alright, [this fixed the build on the Android CI and it's able to run the host tests from the compiler validation suite again](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android-arm64/lastBuild/console). @drodriguez, if you have any input, let us know, going to submit this for the 5.6 branch next.

**Pull request merged**
